### PR TITLE
[IMP] pos_meal_voucher : add information on ticket

### PR DESCRIPTION
**Pull request by legalsylvain**
Add informations on ticket if a new checkbox (meal_voucher_display_info_ticket) is enabled on pos.config.

Note : informations seems mandatory in France, for stores with a sales area exceeding 120 square meters.

ticket_information

### CODE REVIEW

**Review by GabbasovDinar — APPROVED** [review]
Code LGTM!

File: pos_meal_voucher/static/src/js/models.js [hunk 1]
```diff
@@ -39,6 +39,52 @@ odoo.define("pos_meal_voucher.models", function (require) {
...
+
+                current_line += name.slice(0, space_index + 1);
+                name = name.slice(space_index + 1);
+            }
+
+            if (current_line.length) {
+                wrapped.push(current_line);
+            }
+
+            return wrapped;
```
Comment (ivantodorovich) [inline pos_meal_voucher/static/src/js/models.js]:
```suggestion

            if (!this.get_product().meal_voucher_ok) { // TODO: check for config, too?
                return OrderlineSuper.generate_wrapped_product_name.apply(this, arguments);
            }
            const originalDisplayName = this.product.display_name;
            this.product.display_name = "(*) " + originalDisplayName;
            const res = OrderlineSuper.generate_wrapped_product_name.apply(this, arguments);
            this.product.display_name = originalDisplayName;
            return res;
```

There is this hackish alternative that usually just works.
Kind of temporarily monkey-patching to achieve what you want, but always making sure to revert to the original at the end.
Works fine in synchronous functions, never to be used on async functions though. ⚠️

Comment (legalsylvain) [reply pos_meal_voucher/static/src/js/models.js]: Awsome ! I'll take a look on this simplification .

File: pos_meal_voucher/readme/ROADMAP.rst [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
 * Add an option to add subtotal of products that can be paid with meal vouchers,
   on the bill.
 * Prevent to scan twice the same Meal Voucher barcode.
+
+Known limitation
+~~~~~~~~~~~~~~~~
+
+* Due to poor design of Odoo Point of Sale module, the current module
+  overwrite ``generate_wrapped_product_name``. This part will be fixed
+  when the module will be ported in V16.
```
Comment (ivantodorovich) [inline pos_meal_voucher/readme/ROADMAP.rst]:
```suggestion
```

I guess this can be removed now

**Review by ivantodorovich — APPROVED** [review]
Don't forget to squash / cleanup the commits 🙏🏻 

LG

**Comment by legalsylvain** [comment]
commit squashed. Thanks for your review ! 

/ocabot merge patch

**Pull request merged**